superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets/EditSection.tsx

```diff
@@ -0,0 +1,150 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { FC, useMemo } from 'react';
+import { HandlerFunction, styled, t } from '@superset-ui/core';
+import { Typography, Tooltip } from 'src/common/components';
+import { useDispatch } from 'react-redux';
+import Button from 'src/components/Button';
+import { setFilterSetsConfiguration } from 'src/dashboard/actions/nativeFilters';
+import { DataMaskUnit } from 'src/dataMask/types';
+import { WarningOutlined } from '@ant-design/icons';
+import { ActionButtons } from './Footer';
+import { useDataMask, useFilterSets } from '../state';
+import { APPLY_FILTERS_HINT, findExistingFilterSet } from './utils';
+
+const Wrapper = styled.div`
+  display: grid;
+  grid-template-columns: 1fr;
+  align-items: flex-start;
+  justify-content: flex-start;
+  grid-gap: ${({ theme }) => theme.gridUnit}px;
+  background: ${({ theme }) => theme.colors.primary.light4};
+  padding: ${({ theme }) => theme.gridUnit * 2}px;
+`;
+
+const Title = styled(Typography.Text)`
+  color: ${({ theme }) => theme.colors.primary.dark2};
+`;
+
+const Warning = styled(Typography.Text)`
+  font-size: ${({ theme }) => theme.typography.sizes.s}px;
+  & .anticon {
+    padding: ${({ theme }) => theme.gridUnit}px;
+  }
+`;
+
+const ActionButton = styled.div<{ disabled?: boolean }>`
+  display: flex;
+  & button {
+    ${({ disabled }) => `pointer-events: ${disabled ? 'none' : 'all'}`};
+    flex: 1;
+  }
+`;
+
+type EditSectionProps = {
+  filterSetId: string;
+  dataMaskSelected: DataMaskUnit;
+  onCancel: HandlerFunction;
+  disabled: boolean;
+};
+
+const EditSection: FC<EditSectionProps> = ({
+  filterSetId,
+  onCancel,
+  dataMaskSelected,
+  disabled,
+}) => {
+  const dataMaskApplied = useDataMask();
+  const dispatch = useDispatch();
+  const filterSets = useFilterSets();
+  const filterSetFilterValues = Object.values(filterSets);
+  const handleSave = () => {
+    dispatch(
+      setFilterSetsConfiguration(
+        filterSetFilterValues.map(filterSet => {
+          const newFilterSet = {
+            ...filterSet,
+            dataMask: { nativeFilters: { ...dataMaskApplied } },
+          };
+          return filterSetId === filterSet.id ? newFilterSet : filterSet;
+        }),
+      ),
+    );
+    onCancel();
+  };
+
+  const foundFilterSet = useMemo(
+    () =>
+      findExistingFilterSet({
+        dataMaskApplied,
+        dataMaskSelected,
+        filterSetFilterValues,
+      }),
+    [dataMaskApplied, dataMaskSelected, filterSetFilterValues],
+  );
+
+  const hasSimilarFilterSet =
+    foundFilterSet && foundFilterSet.id !== filterSetId;
+
+  return (
+    <Wrapper>
+      <Title strong>{t('Editing filter set:')}</Title>
+      <Title>{filterSets[filterSetId].name}</Title>
```

Comment:
While this isn't yet in the design, I think it would be a good idea to make the title editable when in edit mode